server/db/stats_store.go

```diff
@@ -0,0 +1,329 @@
+	defer cancel()
+	if err := s.rDBx.SelectContext(ctx, &rows, q, args...); err != nil {
+		return nil, fmt.Errorf("failed to run query: %w", err)
+	}
+
+	m := make(map[string]int64, 12)
+	for _, row := range rows {
+		m[row.Type] = row.Count
+	}
+
+	return m, nil
+}
+
+func (s *Store) GetCallsByMonth() (map[string]int64, error) {
+	s.metrics.IncStoreOp("GetCallsByMonth")
+	defer func(start time.Time) {
+		s.metrics.ObserveStoreMethodsTime("GetCallsByMonth", time.Since(start).Seconds())
+	}(time.Now())
+
+	qb := getQueryBuilder(s.driverName).
+		Select("to_char(to_timestamp(startat / 1000), 'YYYY-MM') AS Month, COUNT(*) AS Count")
+	if s.driverName == model.DatabaseDriverMysql {
+		qb = getQueryBuilder(s.driverName).
+			Select("DATE_FORMAT(FROM_UNIXTIME(startat / 1000), '%Y-%m') AS Month, COUNT(*) AS Count")
+	}
+	qb = qb.From("calls").Where(sq.And{
+		sq.Expr("EndAt > calls.StartAt"),
+		sq.Eq{"DeleteAt": 0},
+		sq.GtOrEq{"StartAt": time.Now().AddDate(0, -12, 0).UnixMilli()},
+	}).GroupBy("Month").OrderBy("Month").Limit(12)
+
+	q, args, err := qb.ToSql()
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare query: %w", err)
+	}
+
+	var rows []struct {
+		Month string
+		Count int64
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(*s.settings.QueryTimeout)*time.Second)
+	defer cancel()
+	if err := s.rDBx.SelectContext(ctx, &rows, q, args...); err != nil {
+		return nil, fmt.Errorf("failed to run query: %w", err)
+	}
+
+	m := make(map[string]int64, 12)
+	for i := 0; i < 12; i++ {
+		m[time.Now().AddDate(0, -i, 0).Format("2006-01")] = 0
+	}
+
+	for _, row := range rows {
+		m[row.Month] = row.Count
+	}
+
+	return m, nil
+}
+
+func (s *Store) GetCallsByDay() (map[string]int64, error) {
+	s.metrics.IncStoreOp("GetCallsByDay")
+	defer func(start time.Time) {
+		s.metrics.ObserveStoreMethodsTime("GetCallsByDay", time.Since(start).Seconds())
+	}(time.Now())
+
+	qb := getQueryBuilder(s.driverName).
+		Select("to_char(to_timestamp(startat / 1000), 'YYYY-MM-DD') AS Day, COUNT(*) AS Count")
+	if s.driverName == model.DatabaseDriverMysql {
+		qb = getQueryBuilder(s.driverName).
+			Select("DATE_FORMAT(FROM_UNIXTIME(startat / 1000), '%Y-%m-%d') AS Day, COUNT(*) AS Count")
+	}
+	qb = qb.From("calls").Where(sq.And{
+		sq.Expr("EndAt > calls.StartAt"),
+		sq.Eq{"DeleteAt": 0},
+		sq.GtOrEq{"StartAt": time.Now().AddDate(0, 0, -30).UnixMilli()},
+	}).GroupBy("Day").OrderBy("Day").Limit(30)
+
+	q, args, err := qb.ToSql()
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare query: %w", err)
+	}
+
+	var rows []struct {
+		Day   string
+		Count int64
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(*s.settings.QueryTimeout)*time.Second)
+	defer cancel()
+	if err := s.rDBx.SelectContext(ctx, &rows, q, args...); err != nil {
+		return nil, fmt.Errorf("failed to run query: %w", err)
+	}
+
+	m := make(map[string]int64, 30)
+	for i := 0; i < 30; i++ {
```

Comment:
I don't really mind, but does it matter (30 vs 28 vs 31)?

Reply:
Tracking the last 30 days seemed more straightforward than having to deal with month-by-month cases.